### PR TITLE
fix(payments): coupon Component Overlap

### PR DESCRIPTION
**Pull request by IvoJP**
## Because

- We want the components to not overlap depending on browser/device/language

## This pull request

- refactors templates and updates css to prevent components from floating over each other

## Issue that this pull request solves

Closes: #11481 , #11437 , #11438 , #11440 

## Checklist

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [X] I have added necessary documentation (if appropriate).
- [X] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx [hunk 1]
```diff
@@ -75,78 +75,76 @@ export const CouponForm = ({ planId, coupon, setCoupon }: CouponFormProps) => {
   };
 
   return (
-    <section
+    <div
```
Comment (chenba) [inline packages/fxa-payments-server/src/components/CouponForm/index.tsx:78]: huh does this make a difference wrt styles?

**Comment by chenba** [comment]
image

small issue: there's a border or `<hr>` overlapping just below the total.

**Comment by chenba** [comment]
> small issue: there's a border or `<hr>` overlapping just below the total.

non-blocking.  I was in desktop browser mode but with a narrow-ish window.

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx (hunk 1 above)
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx:78]: not that i can tell? none of our css was selecting on element type, just classes. i imagine there are some inherent differences (obivously) but i think we are pretty explicit with our css and it appears to be rendering correctly on brave (chromium based) and firefox as well across different screen sizes on both.

**Review by chenba — APPROVED** [review]
Looked fine when I tried the various emulation options in Firefox's dev tools. 👍

File: packages/fxa-payments-server/src/components/CouponForm/index.tsx (hunk 1 above)
Comment (chenba) [reply packages/fxa-payments-server/src/components/CouponForm/index.tsx:103]: I was curious why you made that change if it did not make a difference.

**Pull request merged**